api/v1beta1/azureimage_validation_test.go

```diff
@@ -227,3 +227,41 @@ func createTestImageByID(imageID string) *Image {
 		ID: &imageID,
 	}
 }
+
+func TestValidateSingleDetailsOnly(t *testing.T) {
+	g := NewWithT(t)
+
+	// Case 1: image.Marketplace != nil, image details are found
+	image1 := &Image{
+		ID: ptr.To("ID1234"),
+		Marketplace: &AzureMarketplaceImage{
+			ImagePlan: ImagePlan{
+				Offer:     "OFFER",
+				Publisher: "PUBLISHER",
+				SKU:       "SKU",
+			},
+			Version: "1.0.0.",
+		},
+	}
+	errs1 := validateSingleDetailsOnly(image1, field.NewPath("image"))
+	g.Expect(errs1).To(HaveLen(1))
+
+	// Case 2: image.Marketplace == nil, image.ComputeGallery != nil, image details are found
+	image2 := &Image{
+		ID: ptr.To("ID1234"),
+		ComputeGallery: &AzureComputeGalleryImage{
+			Name:           "IMAGENAME",
+			Gallery:        "GALLERY9876",
+			Version:        "1.0.0",
+			SubscriptionID: ptr.To("SUB1234"),
+			ResourceGroup:  ptr.To("RG1234"),
+		},
+	}
+	errs2 := validateSingleDetailsOnly(image2, field.NewPath("image"))
```

Comment:
Can we just reuse the `err` variable instead of making 3 separate ones?
```suggestion
	err = validateSingleDetailsOnly(image2, field.NewPath("image"))
```